nkpremices/EPIC-Mail
# #164804949 Improve project structure and add more validations

#### What does this PR do?
Improve the project structure by adding helpers and middlewares

#### Description of Task to be completed?
Have the previous endpoints working withe the new structure of the project
Have also  additionnal validations on endpoints

#### How should this be manually tested?
After cloning the repo, run the server by the command npm start, then tests the endpoints with potman. Make sure to respect the right json format available on the documentation of the app

#### Any background context you want to provide?
There are some additional validations implemented

#### What are the relevant pivotal tracker stories?
#164430194
#164441516
#164679230
